### PR TITLE
Feature: Audio

### PR DESCRIPTION
**Pull request by ScrappyOrc**
I implemented Audio! You can attach an emitter to an object in the scene, and you can call play( string soundname ) or playFollow( string soundName ) where soundName is the name of the audio file you put into the Audio resources folder (without the file extension). PlayFollow() will play the sound, and move with the gameobject as it moves through the scene, and play() will play the sound from where the gameobject is when play() is called, and won't move with the object. Play() is useful for "burst" type sounds, such as gunfire, and playFollow() is useful for sounds like talking that need to move with the source.

### CODE REVIEW

File: source/dash/components/audio.d [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+				                                  owner.transform.position.z );
+			}
+		}
+	}
+
+
+	/**
+	 * Plays a sound that will follow the emitter for however long you set the length to be.
+	 */
+	void playFollow( string soundName, float soundLength )
```
Comment (dislogical) [inline source/dash/components/audio.d]: Do we still need this overload?

**Comment by dislogical** [comment]
Looks good to me, other than the `playFollow` thing. Also, it looks like you used tabs for spacing, could you convert those to spaces before we merge?

**Comment by wozniakty** [comment]
:+1: super cool

**Comment by dislogical** [comment]
MERGING THA TRAP

**Pull request merged**